File: src/main/java/minseok/cqrschallenge/main/controller/MainController.java
package minseok.cqrschallenge.main.controller;

import lombok.RequiredArgsConstructor;
import minseok.cqrschallenge.common.dto.ApiResponse;
import minseok.cqrschallenge.main.dto.response.MainPageResponse;
import minseok.cqrschallenge.main.service.MainService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/main")
@RequiredArgsConstructor
public class MainController {
    
    private final MainService mainService;
    
    @GetMapping
    public ResponseEntity<ApiResponse<MainPageResponse>> getMainPageData() {
        MainPageResponse response = mainService.getMainPageData();
        
        return ResponseEntity.ok(ApiResponse.success(
                response, 
                "메인 페이지 상품 목록을 성공적으로 조회했습니다."
        ));
    }
}

File: src/modules/main/main.controller.ts
import { Controller, Get } from '@nestjs/common';
import { MainService } from './main.service';
import { GetMainPageResponseDto } from './dto/main.dto';
import { createSuccessResponse } from '~/common/utils/response.util';

@Controller('main')
export class MainController {
  constructor(private readonly mainService: MainService) {}

  @Get()
  async getMainPage(): Promise<GetMainPageResponseDto> {
    const data = await this.mainService.getMainPage();
    return createSuccessResponse(
      data,
      '메인 페이지 상품 목록을 성공적으로 조회했습니다.',
    );
  }
}
